scene/resources/navigation_mesh_source_geometry_data.h

```diff
@@ -28,57 +28,36 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
 /**************************************************************************/
 
-#ifndef NAVIGATION_MESH_GENERATOR_H
-#define NAVIGATION_MESH_GENERATOR_H
+#ifndef NAVIGATION_MESH_SOURCE_GEOMETRY_DATA_H
+#define NAVIGATION_MESH_SOURCE_GEOMETRY_DATA_H
 
-#ifndef _3D_DISABLED
+#include "scene/3d/visual_instance_3d.h"
 
-#include "scene/3d/navigation_region_3d.h"
+class NavigationMeshSourceGeometryData : public Resource {
```

Comment:
Are we intending to store this to disk?  If not, it probably shouldn't be a `Resource` and be `ReferenceCounted` instead.

Reply:
It is indeed a resource with the intention to store it to disk if required. Parsing a large scene can take a while and is not thread-safe in Godot due to how SceneTree and RenderingServer (for accessing mesh data) are involved so very problematic at runtime. Loading a large file with the already parsed data from disk can happen in a thread and has no problems. The baking also has no problems running in a background thread.